daemons/ipa-sam/ipa_sam.c

```diff
@@ -519,8 +529,18 @@ static bool ldapsam_extract_rid_from_entry(LDAP *ldap_struct,
 	}
 
 	if (dom_sid_compare_domain(sid, domain_sid) != 0) {
-		DEBUG(10, ("SID %s is not in expected domain %s\n",
-			   str, sid_string_dbg(domain_sid)));
+		char *debug_domain_sid = NULL;
+		err = sss_idmap_smb_sid_to_sid(idmap_ctx,
+					       discard_const(domain_sid),
+					       &debug_domain_sid);
```

Comment:
Is there a reason why you call sss_idmap_smb_sid_to_sid() here directly instead of sid_talloc_string()? You can use (although not nice) str or sid as memory context since those are freed in the end. Or by adding a tmp_ctx which could also be used to remove the talloc on NULL in get_single_attribute().

I'm asking because having at least sss_idmap_smb_sid_to_sid() only in one place would help to make the transition into the Samba source tree more easy since I don't expect that you want to introduce a dependency on libsss_idmap into Samba.

Reply:
I considered that and decided to keep it as it is. We have `sss_idmap_sid_to_smb_sid()` already in use so this part of the code will anyway be rewritten. I'm willing to keep this one oddity.